apps/trading/components/share-dialog/share-dialog.tsx

```diff
@@ -0,0 +1,177 @@
+import { create } from 'zustand';
+import {
+  AnchorButton,
+  Button,
+  CopyWithTooltip,
+  Dialog,
+  Input,
+  Intent,
+  TradingCheckbox,
+  VegaIcon,
+  VegaIconNames,
+} from '@vegaprotocol/ui-toolkit';
+import { useT } from '../../lib/use-t';
+import { useVegaWallet } from '@vegaprotocol/wallet-react';
+import { useFindReferralSet } from 'apps/trading/client-pages/referrals/hooks/use-find-referral-set';
+import { Links } from 'apps/trading/lib/links';
+import { useMyTeam } from 'apps/trading/lib/hooks/use-my-team';
+import { Link } from 'react-router-dom';
+
+type ShareDialogStore = {
+  isOpen: boolean;
+  withTeam: boolean;
+  setOpen: (isOpen: boolean) => void;
+  setWithTeam: (withTeam: boolean) => void;
+};
+
+export const useShareDialogStore = create<ShareDialogStore>()((set) => ({
+  isOpen: false,
+  withTeam: false,
+  setOpen: (isOpen) => {
+    set({ isOpen });
+  },
+  setWithTeam: (withTeam) => {
+    set({ withTeam });
+  },
+}));
+
+const TWITTER_TWEET_LINK =
+  'https://twitter.com/intent/tweet?text={{TEXT}}&url={{URL}}';
+
+export const ShareDialog = () => {
+  const t = useT();
+  const location = globalThis.location;
+
+  const [isOpen, setOpen] = useShareDialogStore((state) => [
+    state.isOpen,
+    state.setOpen,
+  ]);
+  const [withTeam, setWithTeam] = useShareDialogStore((state) => [
+    state.withTeam,
+    state.setWithTeam,
+  ]);
+
+  const { pubKey } = useVegaWallet();
+
+  const {
+    data: referralSet,
+    loading: referralLoading,
+    role: referralRole,
+  } = useFindReferralSet(pubKey);
+
+  const { team, loading: teamLoading } = useMyTeam();
+
+  let referralCode: string | undefined = undefined;
+
+  const shareLink = new URL(location.origin);
+  shareLink.hash = Links.INVITE();
+
+  if (!referralLoading && referralSet && referralRole === 'referrer') {
+    referralCode = referralSet.id;
+    shareLink.hash = Links.INVITE_REFERRAL_CODE(referralCode);
+  }
+
+  if (!teamLoading && team && withTeam) {
+    shareLink.hash = Links.INVITE_TEAM_CODE(team.teamId);
+    if (referralCode) {
+      shareLink.hash = Links.INVITE_REFERRAL_CODE_AND_TEAM(
+        referralCode,
+        team.teamId
+      );
+    }
+  }
+
+  return (
+    <Dialog
+      open={isOpen}
+      onChange={(isOpen) => setOpen(isOpen)}
+      title={t('SHARE_DIALOG_TITLE')}
+      description={t('SHARE_DIALOG_DESCRIPTION')}
+      size="medium"
+    >
+      <div className="flex flex-col gap-2">
+        <div className="flex flex-col gap-2">
+          {pubKey ? (
```

Comment:
**nitpick (non-blocking)**: if you are not logged in, the dialog looks like this:

<img width="833" alt="Screenshot 2024-09-17 at 11 39 58" src="https://github.com/user-attachments/assets/f0f77247-77b7-4d8e-9ebb-a28856a2765e">

Logging in changes it to make sense:

<img width="791" alt="Screenshot 2024-09-17 at 11 41 37" src="https://github.com/user-attachments/assets/84ba1dcd-8194-4f2a-88b4-875a0445204b">


So, a minor suggestion: Change the label if the user is not logged in, or doesn't have a referral link, to say 'Link' rather than referral link. Or 'Share link', which makes less sense but is clearer

Reply:
I chose the "Share link" option.